### ISSUE TITLE
Hint: Ramda throws DOMException when Cross-origin Window objects are accessed

### ISSUE THREAD

**Issue by semmel** [ramda/ramda#2533]
Demo:
```javascript
// grab an iframe with a different origin than the page
var iframe = $0; 
iframe.contentWindow === iframe.contentWindow // true - so far, so good
R.identical(iframe.contentWindow, iframe.contentWindow); // Uncaught DOMException: 
// ... Blocked a frame with origin ... from accessing a cross-origin frame
```

E.g. this may occur when using iframes as poor-man's webcomponents or when exchanging data via cross-origin [`window.postMessage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) and using Ramda to validate the dispatched `MessageEvent` which contains the cross-origin Window object in the `source` property.

This is because script access to cross-origin Window objects is [very limited](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy#Cross-origin_script_API_access) (just a handful of properties and `.postMessage()`), accessing well-known Symbol keys like `a['@@functional/placeholder']` in Ramda's [`_isPlaceholder(a)`](https://github.com/ramda/ramda/blob/v0.25.0/source/internal/_isPlaceholder.js) which is used by every of Ramda's `curryN*()` functions raises that DOMException.

I don't see a quick fix for this, event when exempting Windows from being placeholders:
```javascript
var _isPlaceholder = function _isPlaceholder(a) {
    return a != null && typeof a === 'object' && !(a instanceof Window)  && a['@@functional/placeholder'] === true;
};

// Will not work because 
iframe.contentWindow instanceof Window // false for cross-origin Windows
```
So in summary I post my findings here just as a *warning* for all Ramda users, or maybe as a motivation to get rid of `@@functional/placeholder` (?)

**Comment by CrossEye** [comment]
I agree that there's no obvious solution to this.

I have some ideas on how to get rid of the placeholder, but it will be some time before we can really think about including them.

**Comment by customcommander** [comment]
Perhaps [`Symbol.for`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/for) could be an option?

Untested:

```javascript
function _isPlaceHolder(a) {
  return a === Symbol.for('@@functional/placeholder');
}
```

**Comment by semmel** [comment]
To support the placeholder (`R.__`) in commutative auto-curried binary functions like `identical` makes no sense.

`identical(a, b) ≡ identical(b, a)`  ⃕  `identical(__, b)(a) ≡ identical(b)(a)`

So one could implement `identical` by a hand-written curry wrapper like this:

```javascript
var identical = (...args) => {
    if (args.length === 2) {
       return Object.is(args[0], args[1]);
    } else {
       return right => Object.is(args[0], right);
    }
 };
identical.length = 2;
```

What is the risk of breaking user code `identical(__, val)`?
